app/src/main/java/pl/mdanilowski/spotted/di/Modules.kt

```diff
@@ -0,0 +1,98 @@
+package pl.mdanilowski.spotted.di
+
+import androidx.room.Room
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import org.koin.android.ext.koin.androidContext
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.dsl.module
+import pl.mdanilowski.spotted.BuildConfig
+import pl.mdanilowski.spotted.data.ApiService
+import pl.mdanilowski.spotted.data.db.SpottedDB
+import pl.mdanilowski.spotted.data.db.dao.CityDao
+import pl.mdanilowski.spotted.data.interactors.DatabaseInteractor
+import pl.mdanilowski.spotted.data.interactors.DatabaseInteractorImpl
+import pl.mdanilowski.spotted.data.interactors.NetworkInteractor
+import pl.mdanilowski.spotted.data.interactors.NetworkInteractorImpl
+import pl.mdanilowski.spotted.repository.CitiesRepository
+import pl.mdanilowski.spotted.repository.CitiesRepositoryImpl
+import pl.mdanilowski.spotted.ui.add.AddMessageViewModel
+import pl.mdanilowski.spotted.ui.dashboard.DashboardViewModel
+import pl.mdanilowski.spotted.ui.details.MessageDetailsViewModel
+import pl.mdanilowski.spotted.ui.enterCity.EnterCityViewModel
+import pl.mdanilowski.spotted.util.*
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+import java.util.concurrent.TimeUnit
+
+val viewModelModule = module(override = true) {
+    viewModel { EnterCityViewModel(get(), get(), get()) }
+    viewModel { DashboardViewModel(get(), get(), get(), get()) }
+    viewModel { AddMessageViewModel(get(), get(), get()) }
+    viewModel { MessageDetailsViewModel(get()) }
+}
+
+val repositoryModule = module(override = true) {
+    single { Room.databaseBuilder(androidContext(), SpottedDB::class.java, SpottedDB.DB_NAME).build() }
+    single<BaseSchedulers> { BaseSchedulersImpl() }
+
+    // Dao
+    factory<CityDao> { get<SpottedDB>().citiesDao() }
+
+    // Interactors
+    factory<DatabaseInteractor> { DatabaseInteractorImpl(get()) }
+    factory<NetworkInteractor> { NetworkInteractorImpl(get(), get()) }
+
+    // Repos
+    factory<CitiesRepository> { CitiesRepositoryImpl(get(), get()) }
+}
+
+val useCaseModule = module(override = true) {
```

Comment:
I'd rename this module to util module, so use cases are stored in useCaseModule

Reply:
Agreed. Tho, there are no use cases for now.